### ISSUE TITLE
Extract completion date thru pipeline

### ISSUE THREAD

**Issue by roll** [opentrials/opentrials#328]
@jessflem has wrote:

> We currently don't have trial completion dates broken out into an individual field (as we do with registration dates) – is this in the pipeline at the moment? If not, can we please add it?

**Comment by vitorbaptista** [comment]
| Registry | Field |
| --- | --- |
| NCT | completion_date_anticipated, completion_date_actual |
| EUCTR | date_of_the_global_end_of_the_trial |
| ICTRP | N/A |

On NCT, there're also `primary_completion_date` and `primary_completion_date_type`. I'm thinking we'll use `completion_date_actual` if it exists, and `completion_date_anticipated` otherwise. @annapowellsmith is that what you've done? cc @opentrials/research

**Comment by annapowellsmith** [comment]
On NCT, for our purposes, I used `primary_completion_date`. I ignored the type, as we were only interested in trials flagged as completed, so they should all have been actual completion dates.

**Comment by vitorbaptista** [comment]
@annapowellsmith Were you able to find some docs on the difference between `completion_date` and `primary_completion_date`? Was looking for it but couldn't find any. In general, do you think we should only use the actual completion dates, ignoring anticipated ones?

**Comment by annapowellsmith** [comment]
There are [some clinicaltrials.gov field definitions here](https://prsinfo.clinicaltrials.gov/definitions.html) (NB I have no idea whether these are definitive). It suggests the primary completion date is for the primary outcome, which makes sense. Without knowing which your users care about, it would probably be ideal to collect both.

**Comment by vitorbaptista** [comment]
@annapowellsmith Thanks! About collecting both, we have all in our internal DB, but don't expose them yet. The problem is that different registries give us different information, so we need to accomodate for them all (as much as we can).

@opentrials/research We need to define what we use for the `completion_date` here on CT.gov and EUCTR. For instance, EUCTR seems to have different meanings for the `date_of_the_global_end_of_the_trial` depending on the trial. For example:

> E.8.8
> Definition of the end of the trial and justification where it is not the last visit of the last subject undergoing the trial:
> End of Trial in a MS of the EU is defined as the time at which it is deemed that sufficient subjects have been recruited and completed the study as stated in the Clinical Trial Application and ethics application in the MS. Poor recruitment (recruiting less than the anticipated number in the CTA) by a MS is not a reason for premature termination but is considered a normal conclusion to the study in that MS. End of Trial in all other participating countries is defined as LSLV.
> https://www.clinicaltrialsregister.eu/ctr-search/trial/2013-002646-36/FI

I also found this:
> Q: When you say global end of trial date, would it mean study follow-up completion or study completion date and not LSLV.
> A: The global end of trial usually corresponds to last patient last visit LSLV.
> https://eudract.ema.europa.eu/docs/training/List%20of%20questions%20from%20V9%20Workshop15November13.pdf

Overall, this needs more analysis by @opentrials/research on what we should extract. If that ends up being too complex, we can extract data from clinicaltrials.gov first, and then move on to the other sources.

**Comment by sarahcebm** [comment]
I have previously asked the EMA for clarification about this. Here is their response:

"Note that the end of trial is specified in the protocol and it corresponds to the LPLV, however the **global end of trial date** is when the last subject in the trial was examined, or received an intervention globally." My emphasis. 

These definitions come from the CT.gov [glossary](https://clinicaltrials.gov/ct2/about-studies/glossary#C)

PRIMARY COMPLETION DATE
The date on which the last participant in a clinical study was examined or received an intervention and that data for the Primary Outcome Measure were collected. Whether the clinical study ended according to the protocol or was terminated does not affect this date.

STUDY COMPLETION DATE
The date on which the final data for a clinical study were collected because the last study participant made the final visit to the study location (that is, "last subject, last visit").

On this basis I equate Study Completion Date with Global End of Trial Date although the definitions are slightly discrepant.

It's a shame that EUCTR doesn't display a field for primary completion date. It is collected on the protocol registration side at EudraCT but not made available on the public facing register.

**Comment by vitorbaptista** [comment]
@sarahcebm So, in this sense, we should extract the `completion_date_actual` from NCT (ignoring the anticipated?), and `date_of_the_global_end_of_the_trial` from EUCTR, right?

I don't know how prevalent they are, but in EUCTR they have a field which allows the company registering the trial to change the meaning of its end date (see `E.8.8` on https://www.clinicaltrialsregister.eu/ctr-search/trial/2013-002646-36/FI`, for example). How to handle those?

**Comment by sarahcebm** [comment]
@vitorbaptista that appears to be end of trial date for the relevant member state. The date_of_the_global_end_of_the_trial is per the EMA definition i.e the last subject globally. So theoretically unless date_of_the_global_end_of_the_trial has been entered, the trial has not been completed at all sites. You will probably find discrepancies where this has been entered for some but not all entries attached to the one record. In this case, I would assume the trial is complete and the registry has not been updated by all member states. 

Re > we should extract the completion_date_actual from NCT (ignoring the anticipated?), and date_of_the_global_end_of_the_trial from EUCTR, right? are we only trying to match CT.gov and EUCTR at this stage? What about registers other than ICTRP e.g ANZCTR?

**Comment by vitorbaptista** [comment]
@sarahcebm Great, I'll update the table at https://github.com/opentrials/opentrials/issues/328#issuecomment-256402853 with the fields we'll extract.

I couldn't find completion date on ICTRP (but might be missing something). For the other registries, it would be great to know what fields we should extract from each, but as we're not displaying their data now because of licensing issues, it's not a priority.

**Comment by vitorbaptista** [comment]
@annapowellsmith FYI, @georgiana-b just finished working on this issue. We're extracting the completion date from NCT and EUCTR according to https://github.com/opentrials/opentrials/issues/328#issuecomment-256402853. I created issue #534 to deal with the other sources.

When we run our processors, our API will start displaying the completion date information. That depends on #523. If all goes well, we should start adding this information by mid/end of the next week. I'll let you know when it's done.

**Comment by Bengoldacre** [comment]
This is an interesting problem. Different people will want different completion dates for different purposes, and any attempt to normalise across registries will be hindered by poor documentation at registries.  I think, if we're going to attempt to normalise (which makes sense for some work) we should expose the underlying data _and_ a normalised field, with clear and linked public documentation of the provenance of the latter?

**Comment by kerfors** [comment]
"expose the underlying data and a normalised field, with clear and linked public documentation of the provenance of the latter" sounds as a great approach for all normalisation applied e.g. for phase. At the same time it makes the JSON objects more complicated if included with the data. An alternativ could be provenance linked to data, https://www.w3.org/TR/dwbp/#provenance
